In [1]:
import pandas as pd

In [2]:
df=pd.read_csv("C:/Users/User/Desktop/cdac advance modules/pml lecture/class  practice/program_dataset/train.csv", parse_dates=['datetime'])
df1=pd.read_csv("C:/Users/User/Desktop/cdac advance modules/pml lecture/class  practice/program_dataset/train.csv", parse_dates=['datetime'])

In [3]:
df['year'] = df['datetime'].dt.year

df['month'] = df['datetime'].dt.month

df['day'] = df['datetime'].dt.day

df['hour'] = df['datetime'].dt.hour

df['weekday']=df['datetime'].dt.weekday

df1['year'] = df1['datetime'].dt.year

df1['month'] = df1['datetime'].dt.month

df1['day'] = df1['datetime'].dt.day

df1['hour'] = df1['datetime'].dt.hour

df1['weekday']=df1['datetime'].dt.weekday

In [4]:
df.drop(['count','registered','datetime'],axis=1,inplace=True)
df1.drop(['count','casual','datetime'],axis=1,inplace=True)

In [5]:
X=df.drop('casual',axis=1)
X1=df1.drop('registered',axis=1)

In [6]:
y=df['casual']
y1=df1['registered']

In [7]:
from sklearn.model_selection import train_test_split 
from xgboost import XGBRegressor

In [8]:
df2=pd.read_csv("test.csv", parse_dates=['datetime'])

In [9]:
df2['year'] = df2['datetime'].dt.year

df2['month'] = df2['datetime'].dt.month

df2['day'] = df2['datetime'].dt.day

df2['hour'] = df2['datetime'].dt.hour

df2['weekday']=df2['datetime'].dt.weekday

In [10]:
df2.drop('datetime',axis=1,inplace=True)

In [17]:
lr_range = [0.001, 0.01, 0.1, 0.2,0.25, 0.3]
n_est_range = [10,20,30,50,100]
md_range = [2,4,6,8,10]

parameters = dict(learning_rate=lr_range,
                  n_estimators=n_est_range,
                  max_depth=md_range)

from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import KFold
kfold = KFold(n_splits=5, random_state=42,shuffle=True)

clf = XGBRegressor(random_state=1211)
cv = GridSearchCV(clf, param_grid=parameters,
                  cv=kfold,scoring='r2')

cv.fit(X1,y1)

print(cv.best_params_)

print(cv.best_score_)

{'learning_rate': 0.1, 'max_depth': 8, 'n_estimators': 100}
0.9621616090731233


casual:-{'learning_rate': 0.1, 'max_depth': 8, 'n_estimators': 100}
0.9359325736245537

registered:-{'learning_rate': 0.1, 'max_depth': 8, 'n_estimators': 100}
0.9621616090731233



In [24]:
clf = XGBRegressor(learning_rate= 0.1, max_depth=8, n_estimators= 100)
clf.fit(X,y)
y_pred_casual = clf.predict(df2)
y_pred_casual[y_pred_casual<0]=0

In [26]:
clf = XGBRegressor(learning_rate= 0.1, max_depth= 8, n_estimators= 100)
clf.fit(X1,y1)
y_pred_registered = clf.predict(df2)
y_pred_registered[y_pred_registered<0]=0

In [27]:
testData=pd.read_csv("C:/Users/User/Desktop/cdac advance modules/pml lecture/class  practice/program_dataset/test.csv")
datetime=testData['datetime']

In [28]:
import numpy as np
count=np.add(y_pred_casual,y_pred_registered)

In [29]:
submit=pd.DataFrame({'datetime':datetime,'count':count})
submit.to_csv("./submit_XSBRegressor.csv",index=False)